lib/rpc/tls.go

```diff
@@ -346,3 +358,47 @@ func upsertPackage(packages pack.PackageService, pkg loc.Locator, archive utils.
 	_, err = packages.UpsertPackage(pkg, reader, pack.WithLabels(labels))
 	return trace.Wrap(err)
 }
+
+func validateCerts(archive utils.TLSArchive, timestamp time.Time) error {
+	clientKeyPair := archive[pb.Client]
+	if err := validateCert(clientKeyPair.CertPEM, timestamp); err != nil {
+		return trace.Wrap(err, "invalid client certificate (%v)",
+			trace.UserMessage(err))
+	}
+	serverKeyPair := archive[pb.Server]
+	if err := validateCert(serverKeyPair.CertPEM, timestamp); err != nil {
+		return trace.Wrap(err, "invalid server certificate (%v)",
+			trace.UserMessage(err))
+	}
+	caKeyPair := archive[pb.CA]
+	if err := validateCert(caKeyPair.CertPEM, timestamp); err != nil {
+		return trace.Wrap(err, "invalid CA certificate (%v)",
+			trace.UserMessage(err))
+	}
+	return nil
+}
+
+func validateCert(pemBytes []byte, timestamp time.Time) error {
+	cert, err := tlsca.ParseCertificatePEM(pemBytes)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if timestamp.Before(cert.NotBefore) {
+		return newCertError("certificate is valid in the future")
```

Comment:
I would include `now` and `cert.NotBefore` in the error for easier troublshooting. Same for the error below.